### PR TITLE
Add text counter to Profile Bio textarea

### LINKED ISSUE #25843: Bio has character limit, but no indicator
### Steps to reproduce the problem

1. Edit bio in profile page
2. The text will be hard cut off at the character limit, but nothing in the UI indicates it has a character limit

### Expected behaviour

Bio should show a character limit similar to the post composer

### Actual behaviour

No character limit is shown

### Detailed description

_No response_

### Specifications

Mastodon 4.1.3

### PR DESCRIPTION
**Pull request by kevintraver**
Fixes #25843

### CODE REVIEW

**Comment by kevintraver** [comment]
Eventually the React CharacterCounter component could be used here with, but that is going to require some refactoring.

**Comment by renchap** [comment]
This could also be done in a generic way with a data attribute (`data-show-max-chars`), and once present it can dynamically add the HTML and uses the `maxlength` attribute to get the max amount of chars and mark the field as an error when reached.

This avoids using React for this, while making it work with other fields.

**Comment by kevintraver** [comment]
That is a good point. Here I’m just not sure an error is necessary since the textarea actually has a limit and therefore the text would get truncated when reaching the limit.

**Comment by renchap** [comment]
Marking the whole field as error is probably not the good call here indeed, as it is limited. But you could have the counter turn red when the limit is reached, so the user understands why they can not enter any more text.

**Comment by kevintraver** [comment]
@renchap Thanks for the suggestion! I've updated the PR to reflect this.

**Comment by kevintraver** [comment]
At some point it might make sense to have the Profile Bio maxlength a setting that can be set at the server level.

File: app/views/settings/profiles/show.html.haml [hunk 1]
```diff
@@ -18,7 +21,13 @@
         = f.input :display_name, wrapper: :with_block_label, input_html: { maxlength: 30, data: { default: @account.username } }
 
       .fields-group
-        = f.input :note, wrapper: :with_block_label, input_html: { maxlength: 500 }
+        - account_note_maxlength = 500
```
Comment (mjankowski) [inline app/views/settings/profiles/show.html.haml:24]: There is an `Account::NOTE_LENGTH_LIMIT` constant now available that this could from.

**Comment by mjankowski** [comment]
Closing from lack of activity/rebase -- if you want to push forward, can you rebase/reopen, address comments, etc?

**Pull request closed without merging**